client/my-sites/stats/stats-subscribers-overview/index.tsx

```diff
@@ -22,64 +22,71 @@ function calculateQueryDate( daysToSubtract: number ) {
 }
 
 // calculate the stats to display in the cards
-function SubscribersOverviewCardStats( subscribersData: SubscribersData[] ) {
-	const getCount = ( index: number ) => {
-		return subscribersData[ index ]?.data[ 0 ]?.subscribers || 0;
+function SubscribersOverviewCardStats( subscribersData: SubscribersData, index: number ) {
+	const getCount = () => {
+		return subscribersData?.data?.[ 0 ]?.subscribers || 0;
 	};
 
-	const overviewCardStats = [
-		{
-			heading: translate( 'Today' ),
-			count: getCount( 0 ),
-		},
-		{
-			heading: translate( '30 days ago' ),
-			count: getCount( 1 ),
-		},
-		{
-			heading: translate( '60 days ago' ),
-			count: getCount( 2 ),
-		},
-		{
-			heading: translate( '90 days ago' ),
-			count: getCount( 3 ),
-		},
-	];
+	let heading;
+	switch ( index ) {
+		case 0:
+			heading = translate( 'Today' );
+			break;
+		case 1:
+			heading = translate( '30 days ago' );
+			break;
+		case 2:
+			heading = translate( '60 days ago' );
+			break;
+		case 3:
+			heading = translate( '90 days ago' );
+			break;
+		default:
+			heading = '';
+			break;
+	}
 
-	return overviewCardStats;
+	const overviewCardStat = {
+		heading,
+		count: getCount(),
+	};
+
+	return overviewCardStat;
 }
 
 const SubscribersOverview: React.FC< SubscribersOverviewProps > = ( { siteId } ) => {
 	const period = 'day';
 	const quantity = 1;
 	const dates = cardIndices.map( calculateQueryDate );
 
-	const { isLoading, isError, subscribersData } = useSubscribersQueries(
-		siteId,
-		period,
-		quantity,
-		dates
-	);
-
-	const overviewCardStats = SubscribersOverviewCardStats( subscribersData );
+	const subscribersQueries = useSubscribersQueries( siteId, period, quantity, dates );
 
 	return (
 		<div className="subscribers-overview highlight-cards">
-			{ isLoading && <div>Loading...</div> }
-			{ isError && <div>Error: Failed to load data.</div> }
-			{ ! isLoading && ! isError && (
-				<div className="highlight-cards-list">
-					{ overviewCardStats.map( ( overviewCardStat ) => (
+			<div className="highlight-cards-list">
+				{ subscribersQueries.map( ( result, index ) => {
+					if ( result.isLoading ) {
+						return <div key={ index }>Loading...</div>;
+					}
+
+					if ( result.isError ) {
+						return <div key={ index }>Error: Failed to load data.</div>;
+					}
+
+					const subscribersData = result.data;
+					const overviewCardStat = SubscribersOverviewCardStats( subscribersData, index );
```

Comment:
```suggestion
					const overviewCardStat = getCountComparisonCardProps( result.data, index );
```

Since `SubscribersOverviewCardStats` is no longer being used to instantiate a React element, let's change the name of the function to be more accurate, like `getCountComparisonCardProps`.